### PR TITLE
Add Dels officer missions

### PR DESCRIPTION
**Pull request by lumbar527**
This PR adds several missions that will offer when you have the Dels start. The first is to report to Delro (the capitol), and then the others are patrol a system (does the system-choosing thing work here?) and to escort a convoy.

### To-do
- [x] Add more missions that will offer after reporting
- [x] Test stuff

### CODE REVIEW

**Comment by lumbar527** [comment]
Would it make sense if a possible mission was to assist the Dardornai fight the Kroom?

File: data/Starts.txt
Comment (1010todd) [inline data/Starts.txt:241]: Calling officers from across 10 systems to participate in what would've been a by-hour defense doesn't make sense. Defense can practically only make use of ships within the system or a jump away at very best (even then you're still arriving a day late which for a battle of few interceptors it would've been over for hours.

This should be something more planned like an escort duty where most ships might've been on patrol duty or something. Or if as defense mission maybe mention why they know of this future raid so they're sending you to defend against it, e.g. the intelligence network found evidence of a raid being planned.
In alternative to the above it could also be a first strike mission where you're sent to destroy a raid fleet being gathered somewhere.

File: data/Starts.txt
Comment (1010todd) [inline data/Starts.txt:204]: Being a continuation of another mission it shouldn't be minor, that will allow possibility of it being blocked by another mission. Probably should be a priority.

File: data/Starts.txt
Comment (1010todd) [inline data/Starts.txt:241]:
```suggestion
	on complete
		payment 100000
		dialog `Officer <first>, thank you for realizing your role in the Kaote Navy. The commander have authorized <payment> bonus for completing this mission.`
```

File: data/Starts.txt
Comment (1010todd) [inline data/Starts.txt:214]:
```suggestion
	to offer
		has "dels officer: report mission available"
	on offer
		clear "dels officer: report mission available"
```

File: data/Starts.txt
Comment (1010todd) [inline data/Starts.txt:217]:
```suggestion
			`	You are forced to accept.`
```
Feel a little weird to just say this. Rather write in that you acknowledge it and prepare for departure or something.

**Review by 1010todd — CHANGES_REQUESTED** [review]
Also this should be moved to Dels Mission.txt, not in Starts.txt.

**Comment by 1010todd** [comment]
> Would it make sense if a possible mission was to assist the Dardornai fight the Kroom?

Maybe as Dardornai request if you're in their territory or some other captain invites you.

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:125]: Should I switch the sides around, or is there something else that would be better?

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:145]: OK, I'll change that.

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:199]: OK, thanks!

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:132]: Oh, right. I forgot to do that. Thanks!

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:136]: OK, thanks!

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:201]: I guess that makes sense. I think I'll just remove it (with #'s) and add some of those missions.

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:164]: My thoughts on this were that there would be multiple potential missions, but I guess you wouldn't get multiple missions offered at once.

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:201]: Thanks, I guess my dialog wasn't really military.

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:174]: Thanks for catching that!

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt]: I changed that to `You prepare to launch.`

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt:129]: I got rid of the choice, should I also get rid of the message color thing?

File: data/Starts.txt
Comment (lumbar527) [reply data/Starts.txt]: I changed it.

**Comment by lumbar527** [comment]
> Also this should be moved to Dels Mission.txt, not in Starts.txt.

I've moved it.

File: data/Starts.txt
Comment (1010todd) [reply data/Starts.txt]: Maybe, have to test in-game to be sure.

File: data/Starts.txt
Comment (1010todd) [reply data/Starts.txt]: Yeah, I don't think it fits.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:803]: Waypoint would probably be better. Stopover means you have to land on the planet which might get tedious after a while.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:806]: There need to be a way to prevent patrol and escort mission from offering at the same time, maybe have the "Report to Delro" mission predetermine the mission you'll get with branch and random that go to action assigning some condition in the offering conversation.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:810]: The commander probably should just say the mission. Maybe they also hand you data chip to log system activity during patrol.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:848]: They won't follow you if they have staying. Also remove heroic so they aren't suicidal.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:851]: `cargo` is number of type of commodity they will carry, there's far less than 50 commodity type exist.

**Review by 1010todd — CHANGES_REQUESTED** [review]

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:720]: OK, I think I just left it when I removed the choice.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:730]: OK, that makes a little more sense.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:731]: I'll fix that.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:803]: The reason I did stopovers was so I could use the planet filter specifications, and I don't know if they work with waypoints.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:806]: Would it also work if they became minor, or had priority removed?

Comment (lumbar527) [reply data/Dels/Dels missions.txt:810]: I added another offer condition set by randomness.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt]: Sorry, but I'm not sure what you mean by that.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt]: Never mind, waypoints work with the specifications.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt]: I made it quite smaller.

**Comment by lumbar527** [comment]
### **Post-Testing:**
### Problems:
- The patrol mission rarely offers, even though I have several of the requirements stacked up.
- Escort ships do not show up on escort list. (fixed)
- The ship-type check thing does not work. It has been removed.
### What works:
- All missions have offered, though the patrol one has only shown up once.

File: data/Dels/Dels missions.txt
Comment (1010todd) [reply data/Dels/Dels missions.txt]: Instead of handing the data pad with instruction maybe the commander should just say what you need to do and hand you data chip to log the patrol instead.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:847]: Since we have `to spawn` now I think maybe there could be occasional ambushing npcs using `random < <somenumber>` node under it for some chance of encountering some resistance.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:817]:
```suggestion
			`	You get back to your ship and prepare for the mission.`

```
This flows better I think.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:808]:
```suggestion
	description `You must patrol <stopovers> to check for Ufulre activity then return to <origin> to report.`

```

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt]: Is `Once you have landed, a Dels commander comes up and says, "Officer <first>, you have been ordered to patrol <waypoints> to check on Ufulre activity."` good?

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:860]: I think I'll change that to `You receive a message reading...`

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:749]: Thanks, I didn't even think of that!

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:716]: OK, that makes sense.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:721]: Would this be good?
`choice`
`        "(Open it)"`
`    "The message reads..."`

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:844]: Thanks, I didn't think of that.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [reply data/Dels/Dels missions.txt:856]: So that's what my error was!

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:832]:
```suggestion
			`	You get back to your ship and prepare for the mission.`
				accept

```

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:901]:
```suggestion

```

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:876]:
```suggestion
		to spawn
			"random" < 20
			"armament deterrence" > 12

```
Since these fleets have chance to spawn Dacar heavy warship maybe lock one of the spawn behind player's fleet strength first so they have less chance of dealing with 2 Dacar at the same time.
Ideally I probably should make the Ufulre fleet more granular and balance out the strength of each variant but that'll be later.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:714]:
```suggestion

```

**Comment by 1010todd** [comment]
> Maybe the `priority` tag makes the convoy mission veto the patrol mission?

It shouldn't matter since each mission have their own condition requirement so they don't get offered at the same time.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:735]:
```suggestion
			``
				decline

```
Fix this too.

**Review by 1010todd — COMMENTED** [review]
Not why it's not offering then. iirc `priority` still allow another `priority` mission to offer at the same time.

File: data/Dels/Dels missions.txt
Comment (lumbar527) [inline data/Dels/Dels missions.txt:829]: I'm starting a conversation here so we can figure out how to get this to offer more.

Comment (lumbar527) [reply data/Dels/Dels missions.txt:829]: I have added the `not "Dels Officer: <other__type> Mission"` with this: https://github.com/lumbar527/Galactic-War/commit/41b6f1c2c1c9c44cdbbb348c2e676e6ead6fa586

Comment (lumbar527) [reply data/Dels/Dels missions.txt:829]: - [x] Works!

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:755]:
```suggestion
			label accept
			branch escort
				random < 50
			label patrol
			action
				"Dels Officer: Patrol Mission" ++
			``
				goto end



			label escort
			action
				"Dels Officer: Escort Mission" ++
			``
				goto end

			label end
			`	You key in the destination and prepare your ship for the mission.`
				accept
```
Empty ending feels a little odd so maybe just a bit of text.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:813]:
```suggestion

```
Apparently this is not necessary.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:906]:
```suggestion
		personality

```

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:849]:
```suggestion

```

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:708]:
```suggestion
	description `You have been ordered to go to <destination> for an assignment. You must reach <planet> by <date>.`

```

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:720]:
```suggestion
	to offer
		"random" < 30 - "Dels Officer: report complete"  # You get called less frequently over time
		has "GW Dels start: officer"
		or
			"ships: Interceptor" > 0
			"ships: Light Warship" > 0
			"ships: Medium Warship" > 0
			"ships: Heavy Warship" > 0
		not "Dels Officer: Report to Delro: active"
		not "Dels Officer: Patrol Systems: active"
		not "Dels Officer: Escort Convoy: active"

```
Probably shouldn't offer while you're doing the job.

**Review by 1010todd — COMMENTED** [review]
Testing the mission atm.

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:864]:
```suggestion
			`Once you have landed, a Dels commander comes up to you and says, "Officer <first>, you must escort a small convoy of three Manoons to <destination> in order to help the people there recover from a recent raid. The ships are currently parked inside navy hangar 5 and will be ready to launch in an hour."`

```

File: data/Dels/Dels missions.txt
Comment (1010todd) [inline data/Dels/Dels missions.txt:865]:
```suggestion
			`	You went to check the freighters you're escorting and agree on the time to launch then back to your ship and prepare for the mission.`

```

**Review by 1010todd — APPROVED** [review]

**Pull request merged**
